Shield/MainPage.xaml.cs

```diff
@@ -607,6 +626,37 @@ public void UpdateDestinations()
                 this.Log("Toast:" + e.Message);
             }
         }
+
+        private static string GetConnectionDetail(Connection connection)
+        {
+            if (null == connection.Source)
+            {
+                return null;
+            }
+
+            if (connection.Source is DeviceInformation)
+            {
+                return (connection.Source as DeviceInformation).Id;
+            }
+            else if (connection.Source is EndpointPair)
+            {
+                var source = connection.Source as EndpointPair;
+                return string.Format("{0}:{1}", source.RemoteHostName, source.RemoteServiceName);
+            }
+            else if (connection.Source is RemotePeer)
+            {
+                var source = connection.Source as RemotePeer;
+                return source.ToString();
+            }
+            else if (connection.Source is string)
+            {
```

Comment:
these can be combined ... the .ToString() operates even if it is already a string